platform/icon/client.go

```diff
@@ -1,39 +1,34 @@
 package icon
 
 import (
-	"encoding/json"
-	"fmt"
 	"github.com/trustwallet/blockatlas/pkg/blockatlas"
-	"github.com/trustwallet/blockatlas/pkg/errors"
-	"net/http"
 	"net/url"
 	"strconv"
 )
 
 type Client struct {
-	HTTPClient *http.Client
-	RPCURL     string
+	blockatlas.Request
 }
 
-func (c *Client) GetAddressTransactions(address string) ([]Tx, error) {
-	uri := fmt.Sprintf("%s/address/txList?%s",
-		c.RPCURL,
-		url.Values{
-			"address": {address},
-			"count":   {strconv.FormatInt(blockatlas.TxPerPage, 10)},
-		}.Encode())
-
-	httpRes, err := c.HTTPClient.Get(uri)
-	if err != nil {
-		return nil, errors.E(err, errors.TypePlatformUnmarshal, errors.Params{"uri": uri})
+func InitClient(baseUrl string) Client {
+	return Client{
+		Request: blockatlas.Request{
```

Comment:
can we create client extension to do this: `Client{url: "google.com"}` to write even smaller inits?

Reply:
https://github.com/trustwallet/blockatlas/pull/412